apps/server/src/modules/tool/service/common-tool.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import {
+	ContextExternalToolDO,
+	ExternalToolDO,
+	SchoolExternalToolDO,
+	ToolConfigurationStatus,
+	ToolVersion,
+} from '@shared/domain';
+
+@Injectable()
+export class CommonToolService {
+	determineToolConfigurationStatus(
+		externalTool: ExternalToolDO,
+		schoolExternalTool: SchoolExternalToolDO,
+		contextExternalTool: ContextExternalToolDO
+	): ToolConfigurationStatus {
+		if (
+			this.compareVersions(schoolExternalTool, externalTool) &&
+			this.compareVersions(contextExternalTool, schoolExternalTool) &&
+			this.compareVersions(contextExternalTool, externalTool)
+		) {
+			return ToolConfigurationStatus.LATEST;
+		}
+
+		return ToolConfigurationStatus.OUTDATED;
+	}
+
+	compareVersions(tool1: ToolVersion, tool2: ToolVersion): boolean {
```

Comment:
I would rename the function, since functions that return a boolean usually answer a 'is' or 'has' question. compare functions usually return {-1, 0, 1} depending on if it is greater, equal or less.